src/wristpy/processing/analytics.py

```diff
@@ -83,14 +85,17 @@ def run_sleep_detection(self) -> List[SleepWindow]:
             A list of SleepWindow instances, each instance contains a sleep onset/wakeup
             time pair.
         """
+        logger.debug("Beginning sleep detection.")
         spt_window = self._spt_window(self.anglez)
         sib_periods = self._calculate_sib_periods(self.anglez)
         spt_window_periods = _find_periods(spt_window)
         sib_window_periods = _find_periods(sib_periods)
         sleep_onset_wakeup = self._find_onset_wakeup_times(
             spt_window_periods, sib_window_periods
         )
-
+        logger.debug(
```

Comment:
For all logging statements with variables in them, avoid f-strings (c.f. https://docs.astral.sh/ruff/rules/logging-f-string/ for an explanation of why). 

Personally, I prefer the *args method, but either *args or extra (both described in the above link) are fine. Optionally, you could enable Ruff's G-rules to be notified of this and several other issues.